asyncscope.md

```diff
@@ -2317,7 +2317,225 @@ spec here
+
+  private:
+    simple_counting_scope* @_scope_@{}; // @_exposition-only_@
+  };
+
+  struct @_join-t_@; // @_exposition-only_@
+
+  enum @_state-type_@ { // @_exposition-only_@
+    @_unused_@, // @_exposition-only_@
+    @_open_@, // @_exposition-only_@
+    @_close_@, // @_exposition-only_@
+    @_open-and-joining_@, // @_exposition-only_@
+    @_closed-and-joining_@, // @_exposition-only_@
+    @_unused-and-closed_@, // @_exposition-only_@
+    @_joined_@, // @_exposition-only_@
+  };
+
+  // [exec.scounting.ctor], constructor and destructor
+  simple_counting_scope() noexcept;
+  simple_counting_scope(simple_counting_scope&&) = delete;
+  ~simple_counting_scope();
+
+  // [exec.scounting.mem], members
+  token get_token() noexcept;
+  void close() noexcept;
+  auto join() noexcept;
+
+private:
+  size_t @_count_@; // @_exposition-only_@
+  @_state-type_@ @_state_@; // @_exposition-only_@
+};
+```
+
+[1]{.pnum} A `simple_counting_scope` maintains a count of outstanding
+    operations.  Let `s` be an object of type `simple_counting_scope`,
+    `t` be an object of type `token` obtained from `s.get_token()`,
+    let `j` be a sender obtained from `s.join()`, and let `o` be
+    an operation state obtained from connecting `j` to a receiver.
+    During its life-time `s` goes through different states which
+    govern what operations are allowed and the result of these
+    operations:
+
+- [1.1]{.pnum} `@_unused_@`: a newly constructed object starts in the
+    `@_unused_@` state.
+- [1.2]{.pnum} `@_open_@`: when `t.try_associate()` is called while
+    `s` is in `@_unused_@` state, `s` moves to the `@_open_@` state.
+- [1.3]{.pnum} `@_open-and-joining_@`: when the operation state `o` is
+    started while the `s` is in `@_unused_@` or `@_open_@` state, `s`
+    moves to the `@_open-and-joining_@` state.
+- [1.4]{.pnum} `@_closed_@`: when `s.close()` is called while `s` is
+    in `open` state, `s` moves to the `@_closed_@` state.
+- [1.5]{.pnum} `@_unused-and-closed_@`: when `s.close()` is called while
+    `s` is in `@_unused_@` state, `s` moves to the `@_unused-and-closed_@`
+    state.
+- [1.6]{.pnum} `@_closed-and-joining_@`: when `s.close()` is called while
+    `s` is in `@_open-and-joining_@` state or the operation state `o` is 
+    started while `s` is in `@_closed_@` or `@_unused-and-close_@` state, `s`
+    moves to the `@_closed-and-joining_@` state.
+- [1.7]{.pnum} `@_joined_@`: when the count of associated objects drops to
+    zero while `s` is in `@_open-and-joining_@` or `@_closed-and-joining_@`
+    state, `s` moves to the `@_joined_@` state.
+
+##### Constructor and Destructor [exec.scounting.ctor]
+
+`simple_counting_scope() noexcept;`
+
+[1]{.pnum} _Postcondtions:_ `@_count_@` is `0` and `@_state_@` is `@_unused_@`
+
+`~simple_counting_scope();`
+
+[2]{.pnum} _Effects:_ If `@_state_@` is not one of `@_joined_@`, `@_unused_@`,
+    or `@_unused-and-close_@`, invokes `terminate` ([except.terminate]{.sref}).
```

Comment:
```suggestion
    or `@_unused-and-closed_@`, invokes `terminate` ([except.terminate]{.sref}).
```

Reply:
I fixed that instance, the other your noticed, and I think I found two more...